sbndqm/dqmAnalysis/CRT/SBND/BernCRTdqmSBND_module.cc

```diff
@@ -136,175 +150,154 @@ sbndaq::BernCRTdqmSBND::~BernCRTdqmSBND()
+  //size_t num_fragments = fragmentHandle->size();
+  //size_t num_hits = hit_vector.size();
   
   //Variables used in Grafana to be sent to DQM OM:
   size_t num_t1_resets = 0;
   size_t hitsperplane[7] = {0,0,0,0,0,0,0};
 
-  //Initialize variables used to calculate pedestals and flag3
-  for(int c=0; c<32; c++){
-    sbndaq::BernCRTdqmSBND::pedSum[c] = 0.; 
-    pedMax[c] = 0.; 
-    ped2Max[c] = 0.; 
-    sbndaq::BernCRTdqmSBND::pedSumSq[c] = 0.; 
-    sbndaq::BernCRTdqmSBND::pedNHits[c] = 0.; 
-    sbndaq::BernCRTdqmSBND::flag3channel[c] = 0.; 
-    //sbndaq::BernCRTdqmSBND::NHits[c] = 0.; 
-  }
-
-  // Initialize variables used to calculate deadtime
-  int count_hit = 0;
-  uint64_t deadtime;
-  //uint64_t & prev_fragment_timestamp = hit.timestamp;
-  uint64_t prev_fragment_timestamp;
-
   // Flag 3 hits
   uint64_t flag3hit = 0;
-  uint64_t missingT0 = 0;
-  uint64_t missingT1 = 0;
-
 
   //loop over all CRT hits in an event
   for(const auto & hit : hit_vector) {
 
-    enum Detector {SIDE_CRT, TOP_CRT};
-//    const Detector detector = IsSideCRT(hit) ? SIDE_CRT : TOP_CRT;
-    const uint16_t & fragment_id        = hit.fragment_ID;
-    /**
-     * TODO:
-     * In order to distinguish between Top and Side CRT
-     * use the variable detector, defined above
-     * Otherwise, MAC address alone is not sufficient,
-     * as some MACs overlap between Top and Side
-     *
-     * Alternative: use fragment_ID directly (fragment_IDs
-     * are unique)
-     */
+    // Extract core hit information
+    const uint8_t& mac5 = hit.mac5;
+
+    // Extract metadata information
+    const uint64_t& fragmentTS = hit.timestamp;
+    const bool isTs0Reset      = hit.IsReference_TS0();
+    const bool isTs1Reset      = hit.IsReference_TS1();
+    const bool ts0Good         = !hit.IsOverflow_TS0();
+    const bool ts1Good         = !hit.IsOverflow_TS1();
 
+    std::string mac5Str = std::to_string(mac5);
+    if(fDebug) std::cout << "Mac5: " << mac5Str <<std::endl;
+
+    enum Detector {SIDE_CRT, TOP_CRT};
+    //const uint16_t & fragment_id        = hit.fragment_ID;
     const uint64_t & fragment_timestamp = hit.timestamp;
 
     //data from FEB:
-    const uint8_t & mac5     = hit.mac5;
-    unsigned readout_number  = mac5;
-    std::string readout_number_str = std::to_string(readout_number);
-
-    //std::cout<<"Mac5: "<<readout_number_str<<std::endl;
+    std::string mac5_str = std::to_string(mac5);
 
     //store the timing and flag information from a hit
-    const int ts0      = hit.ts0;
-    const int ts1      = hit.ts1;
+    const uint32_t ts0      = static_cast<uint32_t>(hit.ts0);
+    const uint32_t ts1      = static_cast<uint32_t>(hit.ts1);
     const bool     isTS0    = hit.IsReference_TS0();
     const bool     isTS1    = hit.IsReference_TS1();
     const bool     isTS0good=!hit.IsOverflow_TS0();
-    const bool     isTS1good=!hit.IsOverflow_TS1();
+    [[maybe_unused]] const bool     isTS1good=!hit.IsOverflow_TS1();
```

Comment:
We're duplicating with the block above? Just with subtly different names.

Reply:
Hmm. True - removing duplicates